MathlibTest/DeprecatedSyntaxLinter.lean

```diff
@@ -77,3 +77,25 @@ warning: declaration uses 'sorry'
 -/
 #guard_msgs in
 example : False := by admit
+
+set_option linter.style.maxHeartbeats true
+/--
+warning: Please, add a comment explaining the need for modifying the maxHeartbeat limit, as in
+set_option maxHeartbeats 10 in
+-- reason for change
+...
+ note: this linter can be disabled with `set_option linter.style.maxHeartbeats false`
+-/
+#guard_msgs in
+set_option maxHeartbeats 10 in
+section
+
+#guard_msgs in
+set_option maxHeartbeats 10 in
+-- no reason, but has a comment
+section
+
+#guard_msgs in
+set_option maxHeartbeats 10 in
+/- no reason, but has a comment -/
```

Comment:
Does this also work with doc-comments? (I don't have an opinion if it should, but a test could be nice :-))

Reply:
No, doc-comments are actually syntax *nodes*, so they are only valid there if they are attached to the following command.

Reply:
Do you think a test for this would be useful?